src/dds/rs-dds-depth-sensor-proxy.cpp

```diff
@@ -0,0 +1,63 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+#include "rs-dds-depth-sensor-proxy.h"
+#include "rs-dds-option.h"
+
+#include <src/librealsense-exception.h>
+#include <rsutils/json.h>
+
+
+namespace librealsense {
+
+
+static const std::string metadata_header_key( "header", 6 );
+static const std::string depth_units_key( "depth-units", 11 );
+
+
+float dds_depth_sensor_proxy::get_depth_scale() const
+{
+    if( auto opt = get_option_handler( RS2_OPTION_DEPTH_UNITS ) )
+    {
+        // We don't want to do a long control-reply cycle on a value that gets updated automatically
+        if( auto dds_option = std::dynamic_pointer_cast< rs_dds_option >( opt ) )
+            return dds_option->get_last_known_value();
+        else
+            return opt->query();
+    }
+    // Rather than throwing an exception, we try to be a little more helpful: without depth units, the depth image will
+    // show black, prompting bug reports. The D400 units min/max are taken from the HW, but the default is set to:
+    return 0.001f;
```

Comment:
Do we have a place for thing like this that are common for D400/D500/DDS devices?

Reply:
No...

Reply:
Then maybe we should create one :-)
Another time...